spec/combine_args_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "spec_helper"
+
+RSpec.describe "#combine_args" do
+  class FakeContainerClass
+    include SitePrism::ElementContainer
+  end
+
+  let(:subject) { FakeContainerClass.new }
+  let(:array_args1) { rand(0..5).times.map { rand(50..200).times.map { ('a'..'z').to_a.sample }.join("") } }
+  let(:array_args2) { rand(0..5).times.map { rand(50..200).times.map { ('a'..'z').to_a.sample }.join("") } }
+  let(:options_args1) { [rand(1..5).times.reduce({}) { |hash, _| hash[rand(5..20).times.map { ('a'..'z').to_a.sample }.join("")] = rand(50..200).times.map { ('a'..'z').to_a.sample }.join(""); hash }] }
+  let(:options_args2) { [rand(1..5).times.reduce({}) { |hash, _| hash[rand(5..20).times.map { ('a'..'z').to_a.sample }.join("")] = rand(50..200).times.map { ('a'..'z').to_a.sample }.join(""); hash }] }
+  let(:array_options_args1) do
+    array = rand(1..5).times.map { rand(50..200).times.map { ('a'..'z').to_a.sample }.join("") }
+    array << rand(1..5).times.reduce({}) { |hash, _| hash[rand(5..20).times.map { ('a'..'z').to_a.sample }.join("")] = rand(50..200).times.map { ('a'..'z').to_a.sample }.join(""); hash }
+    array
+  end
+  let(:array_options_args2) do
+    array = rand(1..5).times.map { rand(50..200).times.map { ('a'..'z').to_a.sample }.join("") }
+    array << rand(1..5).times.reduce({}) { |hash, _| hash[rand(5..20).times.map { ('a'..'z').to_a.sample }.join("")] = rand(50..200).times.map { ('a'..'z').to_a.sample }.join(""); hash }
+    array
+  end
```

Comment:
Use static array options in these let blocks, and hardcoded expected values in the tests (if the same value is used across multiple tests, it is OK to memoize it in a let() block but do not compute the values if at all possible).

These let blocks overcomplicate the tests and also force you to duplicate the production logic in the tests themselves to create the expected result arrays.  Computed expectation values are also not great from a documentation standpoint because you have to run the test and output the resulting expectation to see what it is supposed to be, or read through all of the test/setup code to figure out what the result should be, rather than simply looking at the test and seeing a hardcoded result value.  Computed expectations also make the tests more brittle, difficult to maintain, and more likely that something may break and go undetected.